packager/media/formats/mp4/segmenter.cc

```diff
@@ -214,9 +178,17 @@ Status Segmenter::FinalizeSegment(size_t stream_id,
       first_key_frame = false;
       key_frame_infos_.push_back(
           {key_frame_info.timestamp, moof_start_offset,
-           fragment_buffer_->Size() - moof_start_offset + key_frame_info.size});
+           (fragmenter->fragment()->Size() - fragmenter->data()->Size() 
+            - moof_start_offset + key_frame_info.size)});
     }
-    fragment_buffer_->AppendBuffer(*fragmenter->data());
+    fragment_buffer_->AppendBuffer(*fragmenter->fragment());
+
+    // Generate segment reference.
+    sidx_->references.resize(sidx_->references.size() + 1);
+    fragmenters_[GetReferenceStreamId()]->GenerateSegmentReference(
+        &sidx_->references[sidx_->references.size() - 1]);
+    sidx_->references[sidx_->references.size() - 1].referenced_size +=
```

Comment:
Why is this `+=` ?  What was the default value?  If was 0, then this should just be `=`.  If it was non-zero, then a comment is needed explaining what you're adding to.

Reply:
Oh shoot, this is an artifact from my work when it was in progress. You are completely correct in that it should be just '='. Thank you for catching this!